share/models/base.py

```diff
@@ -43,8 +43,11 @@ class ShareObjectMeta(ModelBase):
     }
 
     def __new__(cls, name, bases, attrs):
-        if (models.Model in bases and attrs['Meta'].abstract) or len(bases) > 1:
-            return super(ShareObjectMeta, cls).__new__(cls, name, bases, attrs)
+        try:
+            if attrs['Meta'].abstract or len(bases) > 1:
+                return super(ShareObjectMeta, cls).__new__(cls, name, bases, attrs)
+        except (KeyError, AttributeError):
```

Comment:
Goooood question... I was trying to give all identifiers a common base class, and I guess I wanted that base class to inherit ShareObject, but only its subclasses get the magic from ShareObjectMeta. This was probably the wrong way to go about it and isn't relevant since I gave up. I'll revert it.